core/src/main/java/org/mskcc/cbio/portal/servlet/QueryBuilder.java

```diff
@@ -502,7 +557,7 @@ private void processData(CohortDetails cohortDetails,
                 if (profileId != null && profileId.length() != 0)
                     geneticProfileMap.put(profileId, GeneticProfileUtil.getProfile(profileId, geneticProfileList));
             }
-            if (cohortDetails.getIsVirtualStudy()) {
+            if (inputStudySampleMap.size()>1 || !inputStudySampleMap.containsKey(singleStudyId)) {
```

Comment:
I'm confused,how could the new condition equels to isVirtualStudy?

Reply:
`inputStudySampleMap`  would be having predefined regular studies and `singleStudyId` is queried id (`request.getAttribute(CANCER_STUDY_ID)`)

over here we check for two conditions
1. if `inputStudySampleMap` has more than one study or
2. if a single virtual study is queried then `inputStudySampleMap`  should not contain that id

is this clear?